contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -577,108 +595,102 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
+            contractAddress: messageInput.feeInfo.contractAddress,
             amount: adjustedFeeAmount
         });
-        sentMessageInfo[destinationChainID][messageID] = SentMessageInfo({
+        sentMessageInfo[messageInput.destinationChainID][
+            messageID
+        ] = SentMessageInfo({
             messageHash: keccak256(teleporterMessageBytes),
             feeInfo: adjustedFeeInfo
         });
 
         emit SendCrossChainMessage(
-            destinationChainID,
+            messageInput.destinationChainID,
             messageID,
             teleporterMessage,
             adjustedFeeInfo
         );
 
         // Submit the message to the AWM precompile.
-        // The Teleporter contract only allows for sending messages to other instances of the same
-        // contract at the same address on other EVM chains, which is why we set the destination adress
-        // as the address of this contract.
         WARP_MESSENGER.sendWarpMessage(teleporterMessageBytes);
 
         return messageID;
     }
 
     /**
-     * @dev Marks the receipt of a message from the given `destinationChainID` with the given `messageID`.
+     * @dev Records the receival of a receipt for a message previsouly sent to the `destinationChainID` with the given `messageID`.
      *
-     * It is possible that the receipt was already received for this message, in which case we return early.
-     * If existing message is found and not yet delivered, we delete it from state and increment the fee/reward
+     * Returns early if a receipt was already previously received for this message, or if the message never existed. Otherwise, deletes
+     * the message information from `sentMessageInfo` and increments the reward redeemable by the specified relayer reward address.
      */
     function _markReceipt(
         bytes32 destinationChainID,
         uint256 messageID,
         address relayerRewardAddress
     ) private {
-        // Get the information about the sent message we are now marking as received.
+        // Get the information about the sent message to be marked as received.
         SentMessageInfo memory messageInfo = sentMessageInfo[
             destinationChainID
         ][messageID];
 
         // If the message hash does not exist, it could be the case that the receipt was already
         // received for this message (it's possible for receipts to be sent more than once)
-        // or that the other chain sent an invalid receipt. We return early since this is an expected
+        // or that the other chain sent an invalid receipt. Return early since this is an expected
         // case where there is no fee to be paid for the given message.
         if (messageInfo.messageHash == bytes32(0)) {
             return;
         }
 
-        // Delete the message information from state now that we know it has been delivered.
+        // Delete the message information from state now that it is known to be delivered.
```

Comment:
nit: "it is known to be" seems a bit awkward. Can we just say "... now that the message has been delivered"?

Reply:
I'd prefer to leave it as because technically the fact that the message is delivered isn't enough to delete it from state here. The sending chain has to be informed of the delivery via a receipt.